x-pack/plugins/infra/server/lib/domains/log_entries_domain/log_entries_domain.ts

```diff
@@ -157,6 +159,13 @@ export class InfraLogEntriesDomain {
             }
           }
         ),
+        context: FIELDS_FROM_CONTEXT.reduce<LogEntry['context']>((ctx, field) => {
+          // Users might have different types here in their mappings.
+          if (typeof doc.fields[field] === 'string') {
+            ctx[field] = doc.fields[field] as string;
+          }
```

Comment:
What do you think of avoiding the explicit type assertion by giving the value a name first so the compiler can do its work? That way it's less likely to silently fail in the future.

```suggestion
          const value = doc.fields[field];
          if (typeof value === 'string') {
            ctx[field] = value;
          }
```

Reply:
I really hope someone at Microsoft is working on this.